src/charm.py

```diff
@@ -642,580 +107,251 @@ def k8s_auth(self) -> None:
 
         self._authed = True
 
-    def _core_v1_api(self) -> kubernetes.client.CoreV1Api:
-        """Use the v1 k8s API.
-
-        Returns:
-            The core v1 API.
-        """
-        self.k8s_auth()
-        return kubernetes.client.CoreV1Api()
-
-    def _networking_v1_api(self) -> kubernetes.client.NetworkingV1Api:
-        """Use the v1 beta1 networking API.
+    @property
+    def _endpoints_controller(self) -> EndpointsController:
```

Comment:
In that case, we need to systematically pass the `namespace` to the controller factory. I don't think that's a significant improvement. Also, I changed all the `*_controller` to be a simple instance attributes instead of property method, since we are moving the `kubernetes.config.load_incluster_config` to the charm class constructor. In that layout, passing the namespace to controller constructor will be difficult.